src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSConnector.java

```diff
@@ -0,0 +1,442 @@
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Assert;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import stepsdesign.BeforeActions;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * GCS Plugin related step design.
+ */
+public class GCSConnector implements CdfHelper {
+
+  static {
+    SeleniumHelper.getPropertiesLocators(GCSLocators.class);
+  }
+
+  List<String> propertiesSchemaColumnList = new ArrayList<>();
+  Map<String, String> sourcePropertiesOutputSchema = new HashMap<>();
+
+  @Given("Open Datafusion Project to configure pipeline")
+  public void openDatafusionProjectToConfigurePipeline() throws IOException, InterruptedException {
+    openCdf();
+  }
+
+  @When("Source is GCS bucket")
+  public void sourceIsGCSBucket() throws InterruptedException {
+    CdfStudioActions.selectGCS();
+  }
+
+  @When("Target is GCS bucket")
+  public void targetIsGCSBucket() {
+    CdfStudioActions.sinkGcs();
+  }
+
+  @When("Target is BigQuery")
+  public void targetIsBigQuery() {
+    CdfStudioActions.sinkBigQuery();
+  }
+
+  @Then("Open GCS Properties")
+  public void openGCSProperties() {
+    CdfStudioActions.clickProperties("GCS");
+  }
+
+  @Then("Connect Source as {string} and sink as {string} to establish connection")
+  public void connectSourceAndSinkToEstablishConnection(String source, String sink) throws InterruptedException {
+    CdfStudioActions.connectSourceAndSink(source, sink);
+  }
+
+  @Then("Enter the GCS Properties with GCS bucket {string} and format {string}")
+  public void enterTheGCSPropertiesWithGCSBucket(String bucket, String format) throws IOException,
+    InterruptedException {
+    CdfGcsActions.enterProjectId();
+    CdfGcsActions.enterReferenceName();
+    CdfGcsActions.getGcsBucket(E2ETestUtils.pluginProp(bucket));
+    CdfGcsActions.selectFormat(E2ETestUtils.pluginProp(format));
+    CdfGcsActions.enterSampleSize(E2ETestUtils.pluginProp("gcsSampleSize"));
+    CdfGcsActions.skipHeader();
+  }
+
+  @Then("Enter the GCS Properties with GCS bucket {string} and format {string} with Path Field {string}")
+  public void enterTheGCSPropertiesWithGCSBucketAndFormatWithPathField(String bucket, String format, String pathField)
+    throws IOException, InterruptedException {
+    enterTheGCSPropertiesWithGCSBucket(bucket, format);
+    CdfGcsActions.enterPathField(E2ETestUtils.pluginProp(pathField));
+  }
+
+  @Then("Enter the GCS Properties with GCS bucket {string} and format {string} " +
+    "with Override field {string} and data type {string}")
```

Comment:
Instead of repeating everything in each step definition can we have separate step definition for each like for eg,
`Enter GCS path {string}`
`Enter GCS format {string}`
`Enter GCS override field {string} and data type {string}`
This will reduce duplication of path and format at every step.

Reply:
Separate steps created for entering properties.